### ISSUE TITLE
Unable to iterate over exec maven plugin

### ISSUE THREAD

**Issue by vdubus** [khmarbaise/iterator-maven-plugin#19]
Hello,

I have a problem with iterator-maven-plugin which cause an error when iterating over [exec-maven-plugin](http://www.mojohaus.org/exec-maven-plugin/). The other plugin isn't able to get dependencies from the POM when it is executed by iterator.

```cmd
mvn clean package

[...]

[INFO] --- iterator-maven-plugin:0.3:iterator (default) @ test-iterator-with-exec ---
[INFO] ------ (1) org.codehaus.mojo:exec-maven-plugin:1.4.0:java
[WARNING]
java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.codehaus.mojo.exec.ExecJavaMojo$1.run(ExecJavaMojo.java:293)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: org/apache/commons/lang3/StringUtils
        at com.vdubus.test.Main.main(Main.java:8)
        ... 6 more
Caused by: java.lang.ClassNotFoundException: org.apache.commons.lang3.StringUtils
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 7 more
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
```
Executing it without iterator does work.

I have created a small project as a showcase (both configurations with and without iterator are in it).

[test-iterator-with-exec.zip](https://mega.nz/#!ZUgVUQSJ!241M6xwJ9L1_4XuSDMnxaBhmMQspHWX82jCnmMbo07I)

**Comment by khmarbaise** [comment]
Unfortunately i can't access the example...You don't have given a correct link to the project? Can make a correct link to the correct project?

**Comment by vdubus** [comment]
Sorry, the GitHub's upload function didn't work.... I changed that, you should be able to download the ZIP now.

**Comment by vdubus** [comment]
Were you able to download the project which show the problem ?

**Comment by AmebaBrain** [comment]
The same thing for me - it cannot parse exec-maven-plugin attributes workingDirectory and executable.

<plugin>
            <groupId>com.soebes.maven.plugins</groupId>
            <artifactId>iterator-maven-plugin</artifactId>
            <version>0.3</version>
            <executions>
               <execution>
                  <phase>pre-integration-test</phase>
                  <goals>
                     <goal>iterator</goal>
                  </goals>
                  <configuration>
                     <items>
                        <item>one</item>
                        <item>two</item>
                        <item>three</item>
                     </items>
                     <pluginExecutors>
                        
                        <pluginExecutor>
                           <plugin>
                              <groupId>org.codehaus.mojo</groupId>
                              <artifactId>exec-maven-plugin</artifactId>
                              <version>1.4.0</version>
                              <executions>
                                 <execution>
                                    <id>run-sqlplus-scripts</id>
                                    <goals><goal>exec</goal></goals>
                                    <phase>pre-integration-test</phase>
                                    <configuration>
                                       <workingDirectory>path_to_scripts </workingDirectory>
                                       <executable>sqlplus</executable>
                                       <arguments>
                                          <argument>-silent</argument>
                                          <argument>@item@</argument>
                                          <argument>@scripts.sql</argument>
                                       </arguments>
                                    </configuration>
                                 </execution>
                              </executions>
                           </plugin>
                        </pluginExecutor>
                     </pluginExecutors>
                  </configuration>
               </execution>
            </executions>
         </plugin>

**Comment by beirtipol** [comment]
have also seen this. Any eta on a fix?

**Comment by garagatyi** [comment]
@khmarbaise Thank you for such a useful plugin!
I need that feature too. I'm using it in Eclipse Che project https://github.com/eclipse/che-lib/blob/master/websocket-terminal/pom.xml

**Comment by khmarbaise** [comment]
Can someone please check the current master if this will solve the problem..My tests say so but i would like to be sure on real life experiences.

**Comment by vdubus** [comment]
@khmarbaise
I have tested it (v0.4-SNAPSHOT) on my sample project above, and it does work now.
I will try to test it on my office project on Monday.

Thanks for your work.

**Comment by vdubus** [comment]
@khmarbaise, I was able to test the SNAPSHOT version of your plugin on my office project, and it works fine.

**Comment by oilid** [comment]
Hey Karl Heinz,

thanks for this great plugin! Could you please release the 0.4 version? This issue is a showstopper for us using your iterator goal...

Thank you very much :-)

Greetz
Oilid

**Comment by rez-** [comment]
This is still an issue and not fixed in 0.5.1
